### PR TITLE
Add ability to register controllers with label selector

### PR DESCRIPTION
**Pull request by saheienko**
**What type of PR is this?**
feature

**What this PR does / why we need it**:
Add ability to register controllers that handle resources with custom labels (server-side filtering).

**Does this PR change a user-facing CRD or CLI?**:

**Is a release note needed?**:

```release-note

```

**Does this change need to be cherry-picked to a release branch?**:

### CODE REVIEW

File: pkg/controller/controller.go [hunk 1]
```diff
@@ -50,13 +50,20 @@ func Run() error {
 	return nil
 }
 
-// Register to get callbacks for updates to objects
-// All handlers need to be registered before calling Run()
-func Register(
+// Register registers a controller with empty label selector.
+func Register(gkv *schema.GroupVersionKind, namespace string, resyncPeriod time.Duration, handler sdk.Handler) error {
```
Comment (disrani-px) [inline pkg/controller/controller.go]: Line it too long with the params, put them on different lines like below

Comment (saheienko) [reply pkg/controller/controller.go]: Updated.

**Comment by disrani-px** [comment]
@saheienko Is this still required?

**Comment by saheienko** [comment]
This is an obsolete one. Closing it.

**Pull request closed without merging**